src/utilities/collection-utils.test.ts

```diff
@@ -572,4 +572,124 @@ describe("CollectionUtils", () => {
     });
 
     // #endregion replaceElementAt
+
+    // -----------------------------------------------------------------------------------------
+    // #region sortByString
+    // -----------------------------------------------------------------------------------------
+
+    describe("#sortByString", () => {
+        type TestType = { letter: string };
+        const selector = (t: TestType) => t.letter;
+
+        it("Sorts an array by specified selector", () => {
+            // Arrange
+            const array = [
+                { letter: "E" },
+                { letter: "C" },
+                { letter: "A" },
+                { letter: "B" },
+                { letter: "D" },
+            ];
+            const expected = [
+                { letter: "A" },
+                { letter: "B" },
+                { letter: "C" },
+                { letter: "D" },
+                { letter: "E" },
+            ];
+
+            // Act
+            const result = CollectionUtils.sortByString(array, selector);
+
+            // Assert
+            expect(result).toEqual(expected);
+        });
+
+        it("Ignores case sensitivity when sorting", () => {
+            // Arrange
+            const array = [
+                { letter: "E" },
+                { letter: "c" },
+                { letter: "A" },
+                { letter: "b" },
+                { letter: "D" },
+            ];
+            const expected = [
+                { letter: "A" },
+                { letter: "b" },
+                { letter: "c" },
+                { letter: "D" },
+                { letter: "E" },
+            ];
+
+            // Act
+            const result = CollectionUtils.sortByString(array, selector);
+
+            // Assert
+            expect(result).toEqual(expected);
+        });
+
+        it("Puts empty string values at the end", () => {
```

Comment:
Could this be reworded to better fit into Given/When/Then style test naming? `given there entries with empty string properties, it returns those values sorted at the end"